File: Examples/VBasic/GroupDocs.Text.Examples.VBasic/Utilities/ExtractText.vb
Imports GroupDocs.Text.Extractors.Text

Public Class ExtractText

    Public Sub New(fileName As String, formatted As Boolean)
        'ExStart:ExtractText
        Dim linesPerPage As Integer = Console.WindowHeight
        Dim factory As New ExtractorFactory()

        Dim extractor As TextExtractor = If(formatted, factory.CreateFormattedTextExtractor(fileName), factory.CreateTextExtractor(fileName))
        If extractor Is Nothing Then
            Console.WriteLine("The document's format is not supported")
            Return
        End If

        Try
            Dim line As String = Nothing
            Do
                Console.Clear()
                Console.WriteLine("{0}", fileName)

                Dim lineNumber As Integer = 0
                Do
                    line = extractor.ExtractLine()
                    lineNumber += 1
                    If line IsNot Nothing Then
                        Console.WriteLine(line)
                    End If
                Loop While line IsNot Nothing AndAlso lineNumber < linesPerPage

                Console.WriteLine()
                Console.WriteLine("Press Esc to exit or any other key to move to the next page")
            Loop While line IsNot Nothing AndAlso Console.ReadKey().Key <> ConsoleKey.Escape
        Finally
            extractor.Dispose()
            'ExEnd:ExtractText
        End Try
    End Sub
    Public Shared Sub ViewContentInConsole(fileName As String)
        'ExStart:ViewContentInConsole
        'get file actual path
        Dim filePath As [String] = Common.getFilePath(fileName)
        Dim extractor As New ExtractText(filePath, filePath.Length > 1 AndAlso filePath = "/f")
        'ExEnd:ViewContentInConsole
    End Sub

End Class

File: Examples/CSharp/GroupDocs.Text.Examples.CSharp/Utilities/ExtractText.cs
using GroupDocs.Text;
using GroupDocs.Text.Extractors.Text;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace GroupDocs.Text_for_.NET.Utilities
{
    public class ExtractText
    {
        public ExtractText(string fileName, bool formatted)
        {
            //ExStart:ExtractText
            int linesPerPage = Console.WindowHeight;
            ExtractorFactory factory = new ExtractorFactory();

            TextExtractor extractor = formatted
                ? factory.CreateFormattedTextExtractor(fileName)
                : factory.CreateTextExtractor(fileName);
            if (extractor == null)
            {
                Console.WriteLine("The document's format is not supported");
                return;
            }

            try
            {
                string line = null;
                do
                {
                    Console.Clear();
                    Console.WriteLine("{0}", fileName);

                    int lineNumber = 0;
                    do
                    {
                        line = extractor.ExtractLine();
                        lineNumber++;
                        if (line != null)
                        {
                            Console.WriteLine(line);
                        }
                    }
                    while (line != null && lineNumber < linesPerPage);

                    Console.WriteLine();
                    Console.WriteLine("Press Esc to exit or any other key to move to the next page");
                }
                while (line != null && Console.ReadKey().Key != ConsoleKey.Escape);
            }
            finally
            {
                extractor.Dispose();
            }
            //ExEnd:ExtractText
        }
        public static void ViewContentInConsole(string fileName)
        {
            //ExStart:ViewContentInConsole
            //get file actual path
            String filePath = Common.GetFilePath(fileName);
            ExtractText extractor = new ExtractText(filePath, filePath.Length > 1 && filePath == "/f");
            //ExEnd:ViewContentInConsole
        }

    }
}
